core/sector_storage/stores/impl/index_impl.cpp

```diff
@@ -0,0 +1,305 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "sector_storage/stores/impl/index_impl.hpp"
+
+#include <boost/filesystem/path.hpp>
+#include <regex>
+#include "primitives/types.hpp"
+#include "primitives/uri_parser/uri_parser.hpp"
+
+namespace fc::sector_storage::stores {
+
+  using fc::primitives::TokenAmount;
+  using fc::primitives::uri_parser::HttpUri;
+  using primitives::sector_file::sectorName;
+  using std::chrono::system_clock;
+
+  std::string toSectorsID(const SectorId &sector_id,
+                          const SectorFileType &file_type) {
+    return sectorName(sector_id) + "_" + toString(file_type);
+  }
+
+  bool isValidUrl(const std::string &url) {
+    static std::regex url_regex(
+        "https?:\\/\\/"
```

Comment:
please leave a reference to the regex source

Reply:
https://stackoverflow.com/questions/38608116/how-to-check-a-specified-string-is-a-valid-url-or-not-using-c-code